### PR TITLE
Add AWS Session Token support

### PR DESCRIPTION
**Pull request by PrasadG193**
This PR:
- Adds support for AWS session token for S3 storage to enable use of temporary credentials
- Adds automated tests to validate session token support

### CODE REVIEW

**Comment by PrasadG193** [comment]
Tested with `kopia` CLI with valid as well as expired session token

**Comment by julio-lopez** [comment]
How do we test this in an automated way?
Could we add tests that only get exercised when a particular variable or credential is set and skipped otherwise? That way, we could create specific roles and credentials and include those in the CI job in the kastenhq/kopia repo.

**Comment by PrasadG193** [comment]
We can configure Minio to support temporary creds (https://docs.min.io/docs/minio-sts-quickstart-guide.html) but it has extra dependencies that needs to be installed

**Comment by PrasadG193** [comment]
Basically we will need to deploy wso2 to achieve that

**Comment by vkamra** [comment]
@PrasadG193 - since we just want to use AssumeRole I think we should be able to just follow directions in https://github.com/minio/minio/blob/master/docs/sts/assume-role.md . Can you try the aws assume-role command list there against play.minio.io ?

**Comment by PrasadG193** [comment]
Sounds good. Will try that

**Comment by PrasadG193** [comment]
@vkamra I was able to generate temporary creds with `aws assume`. I will figure out how to integrate this with existing test workflow

File: repo/blob/s3/s3_storage_test.go [hunk 1]
```diff
@@ -94,6 +114,64 @@ func createBucket(t *testing.T) {
...
 
+func createUser(t *testing.T) {
+	// create minio admin
+	adminCli, err := madmin.New(endpoint, accessKeyID, secretAccessKey, useSSL)
+	if err != nil {
+		t.Fatalf("can't initialize minio admin client: %v", err)
+	}
+
+	// add new kopia user
+	if err = adminCli.AddUser(kopiaUserName, kopiaUserPasswd); err != nil {
```
Comment (vkamra) [inline repo/blob/s3/s3_storage_test.go]: Will this handle user already exists ?

File: repo/blob/s3/s3_storage_test.go [hunk 2]
```diff
@@ -94,6 +114,64 @@ func createBucket(t *testing.T) {
...
+	}
+}
+
+func createTemporaryCreds(t *testing.T) (accessID, secretKey, sessionToken string) {
+	// Configure to use MinIO Server
+	awsConfig := &aws.Config{
+		Credentials:      credentials.NewStaticCredentials(kopiaUserName, kopiaUserPasswd, ""),
+		Endpoint:         aws.String(host),
+		Region:           aws.String("us-east-1"),
+		DisableSSL:       aws.Bool(true),
```
Comment (vkamra) [inline repo/blob/s3/s3_storage_test.go]: Is this required?

File: repo/blob/s3/s3_storage_test.go (hunk 1 above)
Comment (PrasadG193) [reply repo/blob/s3/s3_storage_test.go]: Yes

File: repo/blob/s3/s3_storage_test.go (hunk 2 above)
Comment (PrasadG193) [reply repo/blob/s3/s3_storage_test.go]: Not required. Removed

**Comment by PrasadG193** [comment]
```
$ go test -v ./repo/blob/s3/ -count=1
=== RUN   TestS3Storage
2020/02/11 15:17:06 keeping test-1581414203-e41193b301189bc0-abff4585856ebf0748fd989e1dd623a8963d
2020/02/11 15:17:06 keeping test-1581414203-e41193b301189bc0-abgc3dca496d510f492c858a2df1eb824e62
2020/02/11 15:17:06 keeping test-1581414203-e41193b301189bc0-kopia.repository
2020/02/11 15:17:06 keeping test-1581414203-e41193b301189bc0-zxce0e35630770c54668a8cfb4e414c6bf8f
2020/02/11 15:17:06 keeping test-1581414241-23deabc72a45d178-abff4585856ebf0748fd989e1dd623a8963d
2020/02/11 15:17:06 keeping test-1581414241-23deabc72a45d178-abgc3dca496d510f492c858a2df1eb824e62
2020/02/11 15:17:06 keeping test-1581414241-23deabc72a45d178-kopia.repository
2020/02/11 15:17:06 keeping test-1581414241-23deabc72a45d178-zxce0e35630770c54668a8cfb4e414c6bf8f
2020/02/11 15:17:08 progress abcdbbf4f0507d054ed5a80a5b65086f602b: 0/0
2020/02/11 15:17:09 progress abcdbbf4f0507d054ed5a80a5b65086f602b: 0/0
2020/02/11 15:17:09 progress zxce0e35630770c54668a8cfb4e414c6bf8f: 0/1
2020/02/11 15:17:10 progress zxce0e35630770c54668a8cfb4e414c6bf8f: 1/1
2020/02/11 15:17:11 progress abff4585856ebf0748fd989e1dd623a8963d: 0/1000
2020/02/11 15:17:12 progress abff4585856ebf0748fd989e1dd623a8963d: 1000/1000
2020/02/11 15:17:14 progress abgc3dca496d510f492c858a2df1eb824e62: 0/10000
2020/02/11 15:17:15 progress abgc3dca496d510f492c858a2df1eb824e62: 10000/10000
2020/02/11 15:17:17 progress kopia.repository: 0/100
2020/02/11 15:17:18 progress kopia.repository: 100/100
--- PASS: TestS3Storage (32.89s)
=== RUN   TestS3StorageWithSessionToken
2020/02/11 15:17:38 created session token with assume role: expiration: 2020-02-11 10:02:38 +0000 UTC
2020/02/11 15:17:41 keeping test-1581414203-e41193b301189bc0-abff4585856ebf0748fd989e1dd623a8963d
2020/02/11 15:17:41 keeping test-1581414203-e41193b301189bc0-abgc3dca496d510f492c858a2df1eb824e62
2020/02/11 15:17:41 keeping test-1581414203-e41193b301189bc0-kopia.repository
2020/02/11 15:17:41 keeping test-1581414203-e41193b301189bc0-zxce0e35630770c54668a8cfb4e414c6bf8f
2020/02/11 15:17:41 keeping test-1581414241-23deabc72a45d178-abff4585856ebf0748fd989e1dd623a8963d
2020/02/11 15:17:41 keeping test-1581414241-23deabc72a45d178-abgc3dca496d510f492c858a2df1eb824e62
2020/02/11 15:17:41 keeping test-1581414241-23deabc72a45d178-kopia.repository
2020/02/11 15:17:41 keeping test-1581414241-23deabc72a45d178-zxce0e35630770c54668a8cfb4e414c6bf8f
2020/02/11 15:17:41 keeping test-1581414426-5712ad4781aad8dd-abff4585856ebf0748fd989e1dd623a8963d
2020/02/11 15:17:41 keeping test-1581414426-5712ad4781aad8dd-abgc3dca496d510f492c858a2df1eb824e62
2020/02/11 15:17:41 keeping test-1581414426-5712ad4781aad8dd-kopia.repository
2020/02/11 15:17:41 keeping test-1581414426-5712ad4781aad8dd-zxce0e35630770c54668a8cfb4e414c6bf8f
2020/02/11 15:17:43 progress abcdbbf4f0507d054ed5a80a5b65086f602b: 0/0
2020/02/11 15:17:44 progress abcdbbf4f0507d054ed5a80a5b65086f602b: 0/0
2020/02/11 15:17:45 progress zxce0e35630770c54668a8cfb4e414c6bf8f: 0/1
2020/02/11 15:17:46 progress zxce0e35630770c54668a8cfb4e414c6bf8f: 1/1
2020/02/11 15:17:46 progress abff4585856ebf0748fd989e1dd623a8963d: 0/1000
2020/02/11 15:17:47 progress abff4585856ebf0748fd989e1dd623a8963d: 1000/1000
2020/02/11 15:17:49 progress abgc3dca496d510f492c858a2df1eb824e62: 0/10000
2020/02/11 15:17:50 progress abgc3dca496d510f492c858a2df1eb824e62: 10000/10000
2020/02/11 15:17:53 progress kopia.repository: 0/100
2020/02/11 15:17:54 progress kopia.repository: 100/100
--- PASS: TestS3StorageWithSessionToken (36.74s)
PASS
ok      github.com/kopia/kopia/repo/blob/s3     69.650s
```

**Comment by vkamra** [comment]
LGTM @PrasadG193.

**Comment by vkamra** [comment]
@PrasadG193 - please go ahead and merge with a descriptive commit message

File: repo/blob/s3/s3_storage_test.go [hunk 3]
```diff
@@ -11,7 +11,12 @@ import (
 	"testing"
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (julio-lopez) [inline repo/blob/s3/s3_storage_test.go]: Are there minio equivalent packages and structs for these?

**Comment by julio-lopez** [comment]
LG overall. It'd be good to avoid the additional dependency on the AWS SDK (for now).

File: repo/blob/s3/s3_storage_test.go (hunk 3 above)
Comment (PrasadG193) [reply repo/blob/s3/s3_storage_test.go]: @julio-lopez we are leveraging `aws assume-role` to create temporary creds. For that we need to use aws sdks.

**Pull request merged**